kernel/posix/pthread_common.c

```diff
@@ -18,3 +19,34 @@ void ready_one_thread(_wait_q_t *wq)
 		_ready_thread(th);
 	}
 }
+
+s64_t timespec_to_timeoutms(const struct timespec *abstime)
+{
+	s64_t milli_secs;
+	s32_t secs;
+	struct timespec curtime;
+
+	/* FIXME: Zephyr does have CLOCK_REALTIME to get time.
+	 * As per POSIX standard time should be calculated wrt CLOCK_REALTIME.
+	 * Zephyr deviates from POSIX 1003.1 standard on this aspect.
+	 */
+	clock_gettime(CLOCK_MONOTONIC, &curtime);
+	secs = abstime->tv_sec - curtime.tv_sec;
+
+	if (abstime->tv_sec < curtime.tv_sec ||
+	    (secs == 0 && abstime->tv_nsec <= curtime.tv_nsec)) {
+		milli_secs = K_NO_WAIT;
+	} else {
+		milli_secs = (abstime->tv_nsec - curtime.tv_nsec) /
+			      NSEC_PER_MSEC;
+
+		if (milli_secs < 0) {
```

Comment:
I see the intent here, but this whole if() block is a noop.  You get the same answers if you just remove the 4 lines.  Think about it. :)

Reply:
yes we can get rid of this. this is redundant.